code/drasil-printers/Language/Drasil/DOT/Print.hs

```diff
@@ -0,0 +1,205 @@
+by using subgraphs. The 'dot' method of displaying graphs naturally groups subgraphs.
+-}
+
+type Colour = String
+
+data GraphInfo = GI {
+    assumpLabels :: [String] -- Assumption
+    , ddLabels   :: [String] -- Data definition
+    , gdLabels   :: [String] -- General definition
+    , tmLabels   :: [String] -- Theory model
+    , imLabels   :: [String] -- Instance model
+    , rLabels    :: [String] -- Requirements. Currently cannot differentiated Functional and Non-Functional ones.
+    , gsLabels   :: [String] -- Goal statements
+    , cLabels    :: [String] -- Changes. Currently cannot differentiated Likely and Unlikely ones.
+
+    , directionsAvsA     :: [(String, [String])] -- graph directions
+    , directionsAvsAll   :: [(String, [String])]
+    , directionsRefvsRef :: [(String, [String])]
+    , directionsAllvsR   :: [(String, [String])]
+    , directionsAllvsAll :: [(String, [String])]
+
+    , assumpColour :: Colour -- give the ability to change colours of bubbles within the graph
+    , ddColour     :: Colour
+    , gdColour     :: Colour
+    , tmColour     :: Colour
+    , imColour     :: Colour
+    , rColour      :: Colour
+    , gsColour     :: Colour
+    , cColour      :: Colour
+    
+    -- may need more information regarding ranking & ordering, but for now I'm just keeping it simple
+}
+
+-- Does this need to know a filepath? How can we put it in the same place as the build files?
+-- Is this a drasil-generator type of thing?
+-- Output each graph individually
+outputDot :: FilePath -> GraphInfo -> IO ()
+outputDot outputFilePath gi = do
+    createDirectoryIfMissing False outputFilePath
+    setCurrentDirectory outputFilePath
+    mkOutputAvsA gi
+    mkOutputAvsAll gi
+    mkOutputRefvsRef gi
+    mkOutputAllvsR gi
+    mkOutputAllvsAll gi
+
+mkOutputAvsA :: GraphInfo -> IO ()
+mkOutputAvsA gi = do
+    handle <- openFile "avsa.dot" WriteMode
+    hPutStrLn handle "digraph avsa {"
+    outputSubAvsA gi handle
+    hPutStrLn handle "}"
+    hClose handle
+
+outputSubAvsA :: GraphInfo -> Handle -> IO ()
+outputSubAvsA gi handle = do
+    mapM_ (mkDirections handle) (directionsAvsA gi)
+
+    let labels = filterAndGI gi [assumpLabels]
+        prefixLabels = ["A"]
+        colours = [assumpColour gi]
+
+        allLabels = zip3 labels prefixLabels colours
+
+    hPutStrLn handle "\n"
+    mapM_ (mkNodes handle) allLabels
+
+mkOutputAvsAll :: GraphInfo -> IO ()
+mkOutputAvsAll gi = do
+    handle <- openFile "avsall.dot" WriteMode
+    hPutStrLn handle $ "digraph avsall {"
+    outputSubAvsAll gi handle
+    hPutStrLn handle "}"
+    hClose handle
+
+outputSubAvsAll :: GraphInfo -> Handle -> IO ()
+outputSubAvsAll gi handle = do
+    mapM_ (mkDirections handle) (directionsAvsAll gi)
+
+    let labels = filterAndGI gi [assumpLabels, ddLabels, tmLabels, gdLabels, imLabels, rLabels, cLabels]
+        prefixLabels = ["A", "DD", "TM", "GD", "IM", "R", "C"]
```

Comment:
I don't like these labels being hard-coded here (and duplicated in the file). These should be grabbed from the abbreviation of some chunk(s) instead.